checkov/terraform/graph_builder/variable_rendering/renderer.py

```diff
@@ -298,32 +302,43 @@ def _render_dynamic_blocks(self) -> None:
                     self.local_graph.update_vertex_config(vertex, changed_attributes)
 
     @staticmethod
-    def _process_dynamic_blocks(dynamic_blocks: list[dict[str, Any]]) -> dict[str, list[dict[str, Any]]]:
+    def _process_dynamic_blocks(dynamic_blocks: list[dict[str, Any]] | dict[str, Any])\
+            -> dict[str, list[dict[str, Any]]]:
         rendered_blocks: dict[str, list[dict[str, Any]]] = {}
+        dynamic_type = DYNAMIC_BLOCKS_LISTS
 
         if not isinstance(dynamic_blocks, list):
-            logging.info(f"Dynamic blocks found, but of type {type(dynamic_blocks)}")
-            return rendered_blocks
+            dynamic_blocks = [dynamic_blocks]
+            dynamic_type = DYNAMIC_BLOCKS_MAPS
+            logging.info(f"Dynamic blocks found, with type {type(dynamic_blocks)}")
 
         for block in dynamic_blocks:
             block_name, block_values = next(iter(block.items()))  # only one block per dynamic_block
             block_content = block_values.get("content")
             dynamic_values = block_values.get("for_each")
             if not block_content or not dynamic_values:
-                return rendered_blocks
+                continue
 
             dynamic_value_ref = f"{block_name}.value"
-            dynamic_arguments = [
-                argument
-                for argument, value in block_content.items()
-                if value == dynamic_value_ref
-            ]
+            dynamic_arguments = []
+            for argument, value in block_content.items():
+                if dynamic_type == DYNAMIC_BLOCKS_LISTS and value == dynamic_value_ref:
+                    dynamic_arguments.append(argument)
+                if dynamic_type == DYNAMIC_BLOCKS_MAPS and isinstance(value, str) and dynamic_value_ref in value:
+                    dynamic_arguments.append(argument)
+
             if dynamic_arguments:
                 block_confs = []
                 for dynamic_value in dynamic_values:
                     block_conf = deepcopy(block_content)
                     for dynamic_argument in dynamic_arguments:
-                        block_conf[dynamic_argument] = dynamic_value
+                        if dynamic_type == DYNAMIC_BLOCKS_MAPS:
+                            if not isinstance(dynamic_value, dict):
+                                continue
```

Comment:
It is possible that `dynamic_value` would be a list? then I'd expect to recursively process the block as a dynamic block of list type. If that's a future work, please comment a TODO so it won't be missed